### DISCUSSION TITLE
can we install both k3s Server and agent in single machine or vm?

### DISCUSSION THREAD

**Discussion by Franky45** [k3s-io/k3s#9786] [Q&A] [answered]
Hi,
Environment :  Ubuntu os 22.04  
for installation the command i have used - curl -sfL https://get.k3s.io | sh -
iam trying to install both k3s server as well as agent on single machine.
K3s server is installed successfully  as systemctl status command shows it running.
but during k3s agent installation the command i have used is -  curl -sfL https://get.k3s.io | K3S_URL=https://serverip:6443 K3S_TOKEN=servertoken sh - 
the systemctl status command for agent shows failed why?

is it possible to install both agent and server in single system if yes then provide reference to installation steps

Thank you.

**Comment by brandond** [comment, marked as answer]
No. A node can be either a server or an agent. The k3s server also includes all the agent functionality - container runtime, kubelet, and cni. 

See the diagram at https://k3s.io/#:~:text=How%20it%20Works

**Reply by Franky45** [reply]
Thank you
